posts/2021/03/hello-hpy.md

```diff
@@ -0,0 +1,344 @@
+<!--
+.. title: Hello, HPy
+.. slug: hello-hpy
+.. date: 2021-03-29 10:00:00 UTC
+.. author: Antonio Cuni
+.. tags: 
+.. category: 
+.. link: 
+.. description: 
+.. type: text
+-->
+
+Welcome to the shiny new HPy blog!
+
+HPy has been around for a while now. The initial discussion started during
+EuroPython 2019, at the good old times when we could still go to conferences
+and have real-life meetings. Since then, HPy progressed **a lot** from the
+point of view of the actual code, but we have been a bit too silent
+w.r.t. communicating what we are doing to the external world and to the broader
+Python community. Hopefully, now that this blog is online we will do a better
+job at periodically communicating the status of HPy, so make sure to
+[subscribe to the RSS feed](/rss.xml).
+
+<!--TEASER_END-->
+
+What is HPy?
+============
+
+Quoting the frontpage of our website:
+
+> HPy provides a new API for extending Python in C. In other words, you use
+> `#include <hpy.h>` instead of `#include <Python.h>`.
+
+The official [Python/C API](https://docs.python.org/3/c-api/index.html>) is
+specific to the current implementation of CPython: it exposes a lot of
+internal details which makes it hard:
+
+  - to implement it for other Python implementations (e.g. PyPy, GraalPython,
+    Jython, IronPython, etc.)
+
+  - to experiment with new things inside CPython itself: e.g. using a GC
+    instead of refcounting, or to remove the GIL.
+
+Over the years, it has become evident that
+[emulating the Python/C API in an efficient way is challenging](https://www.pypy.org/posts/2018/09/inside-cpyext-why-emulating-cpython-c-8083064623681286567.html),
+if not impossible. The main goal of HPy is provide a C API which is possible
+to implement in an efficient way on a number of very diverse
+implementations.
+
+There are several advantages to write your C extension in HPy:
+
+  - it runs much faster on PyPy, and at native speed on CPython
+
+  - it is possible to compile a single binary which runs unmodified on all
+    supported Python implementations and versions
```

Comment:
Won't this require coordination with the packaging people to create a tag for universal HPy wheels?

Reply:
we can *compile* a binary even now. To *distribute* the binary yes, it needs cooperation with pypi/wheel/pip/whatever